src/components/List.js

```diff
@@ -51,6 +54,48 @@ const List = ({ firestore }) => {
     setInputText('');
   };
 
+  //* Created two lists
+  const inactiveList = [];
+  const activeList = [];
+
+  //* Used this function to divide active and inactive items
+  filteredList.map(item => {
+    let formatedLastPurchaseDate = dayjs.unix(
+      item.lastPurchasedDate['seconds'],
+    );
+    let formatedToday = dayjs();
+    let difference = formatedToday.diff(formatedLastPurchaseDate, 'd');
+    if (
+      item.numberOfPurchases <= 1 ||
+      parseInt(difference) >= item.nextPurchase * 2
+    ) {
+      inactiveList.push(item)
+    } else {
+      activeList.push(item)
+    }
+  });
+
+
+ //* Used this function to sort both the lists
+  const lists = (arr) => {
+    return arr.sort((a, b) => {
+      if (a.nextPurchase === b.nextPurchase) {
+        return a.name.localeCompare(b.name);
+      }
+      return a.nextPurchase > b.nextPurchase ? 1 : -1;
+    });
+  }
+
+  const active = lists(activeList);
+  const inactive = lists(inactiveList);
+
+  //* Added active and inactive status to the items
+  active.map(a => a['status'] = 'active')
```

Comment:
I might be a bit confused about why the map method is working. I thought map creates a new array rather than mutating the original array, so thought you'd need to store the result of the map method in a new variable (which then those new variables would later be concatenated on line 97). Maybe we can discuss tomorrow!